dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataOkHttpUploader.kt

```diff
@@ -35,10 +36,16 @@ internal class DataOkHttpUploader(
     override fun upload(
         context: DatadogContext,
         batch: List<RawBatchEvent>,
-        batchMeta: ByteArray?
+        batchMeta: ByteArray?,
+        previousUploadStatus: UploadStatus?
```

Comment:
> [!WARNING]
> It feels like we're making a whole lot of change in unrelated classes to simply get the number of attempts for a given batch. 
> If in the end we just keep the info in a `previousUploadStatus` property, why don't we put that info in the `DataOkHttpUploader` class directly, instead of the `DataUploadRunnable`? 

> [!IMPORTANT]
> Also there's a key thing here that we don't track is which batch was the previous upload status related to. Technically, we could make 13 attempts with batch A. Then batch A is deleted for being too old, and we make one successful attempt with batch B, which would report 13 retries… Is this the behavior we want to track ?

Reply:
yup here is a corner case and having followed the exact implementation in iOS I think they have the same corner case. I think the best way in this case is to count the attempt count and `previousFailureCode` based on the requestId in the OkHttpUploader. I am going to open the discussion in the daily today to make sure everyone is aware.

Reply:
We also can leave it here and use the `batchID` to match it.

Reply:
We have 2 options here, either keep the information in the `UploadStatus` and handle the `previousUpload` status by its `batchID` or move everything into the `DataOkHttpUploader` but I would rather not do it there as we do not have any information about the batch id and we will have to compute the fingerprint from the `batchData` or pass the `batchID` as a extra parameter to that method.

Reply:
actually no - scratch above, we don't keep in memory. the deletion of batch for other than successful upload can be a problem here.

Reply:
I would suggest save the batch identifier in the upload status just like retry information and only propagate the information when the identifier is same - else reset it.

I don't see any reason to keep this per batch, only for the active batch is sufficient.

Reply:
I agree with this solution, this will mean that we can keep same implementation and just pass the `batchID` in the previous `UploadStatus` to match it here.